BusinessAdapter.FP/Parsing/FpFileName.cs

```diff
@@ -66,13 +66,25 @@ public static FpFileName Parse(string filename)
 
 		if (isNumeric)
 		{
-			version = parts[4];
+			if (!int.TryParse(parts[4], out int versionInt))
+			{
+				throw new FormatException("Version number in filename is not numeric");
+			}
+			version = versionInt.ToString();
 			timestamp = parts.Length > 6 ? parts[6] : null;
 			messageTypePart = parts[parts.Length - 2];
 		}
 		else
 		{
 			version = parts.Length > 5 ? parts[4] : null;
+			if (version != null)
+			{
+				if (!int.TryParse(parts[4], out int versionInt))
+				{
+					throw new FormatException("Version number in filename is not numeric");
```

Comment:
wollen wir wirklich immer harte Exceptions werfen?

Reply:
In dem Fall könnten wir mit der Datei nicht weiter machen, also muss der Verarbeitungsvorgang abgebrochen werden für diese Datei.